staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -91,27 +111,34 @@ func WithMaxInFlightLimit(
 
 		if c == nil {
 			handler.ServeHTTP(w, r)
-		} else {
+			return
+		}
+
+		if maxInFlightChaosChance != 0 && rand.Float64() < maxInFlightChaosChance {
```

Comment:
if I am not mistaken, the first condition does not help here. Even with floating points 0.0 will never be smaller than 0.0.